### PR TITLE
Move inline docs to online documentation

### LINKED ISSUE #39414: [BITV] Extract documentation iframe and replace it with links
This is accessibility issue.

As decided we have to extract documentation from embedded instance (via iframe) and replace all documentation with links which would forward into nextcloud documentation page.

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/server/issues/39414

This is best reviewed like this: https://github.com/nextcloud/server/pull/40784/files?w=1

- [x] needs https://github.com/nextcloud/documentation/pull/11182

| Before | After |
|---|---|
| image | image |

### CODE REVIEW

File: apps/settings/templates/help.php [hunk 1]
```diff
@@ -41,8 +41,3 @@
 			</a>
 		</li>
 </div>
-
-<div id="app-content" class="help-includes">
-	<iframe src="<?php print_unescaped($_['url']); ?>" class="help-iframe" tabindex="0">
```
Comment (szaimen) [inline apps/settings/templates/help.php:46]: maybe we should keep this and show just a general html file and say that you should choose on the sidebar?

**Review by JuliaKirschenheuter — COMMENTED** [review]
Looks good!
Would `server/28/go.php?to=admin` and `server/28/go.php?to=user` be fixed in a separate issue, right?

**Comment by szaimen** [comment]
> Would `server/28/go.php?to=admin` and `server/28/go.php?to=user` be fixed in a separate issue, right?

Yes, implemented in https://github.com/nextcloud/documentation/pull/11182. Only needs to get deployed now

**Comment by szaimen** [comment]
Setting this back to WIP because there shall be a config switch to bring back the old iframe version

File: apps/settings/appinfo/routes.php [hunk 2]
```diff
@@ -77,7 +77,7 @@
 		['name' => 'TwoFactorSettings#update', 'url' => '/settings/api/admin/twofactorauth', 'verb' => 'PUT' , 'root' => ''],
 		['name' => 'AISettings#update', 'url' => '/settings/api/admin/ai', 'verb' => 'PUT' , 'root' => ''],
 
-		['name' => 'Help#help', 'url' => '/settings/help/{mode}', 'verb' => 'GET', 'defaults' => ['mode' => ''] , 'root' => ''],
+		['name' => 'Help#help', 'url' => '/settings/help', 'verb' => 'GET', 'root' => ''],
```
Comment (szaimen) [inline apps/settings/appinfo/routes.php:80]: revert

File: apps/settings/css/help.css [hunk 3]
```diff
@@ -1,16 +1,32 @@
-.help-includes {
-	overflow: hidden !important;
-}
-
 .help-list__text {
 	margin-left: 24px;
 }
 
-.help-iframe {
-	width: 100%;
```
Comment (szaimen) [inline apps/settings/css/help.css:10]: bring back

File: apps/settings/lib/Controller/HelpController.php [hunk 4]
```diff
@@ -79,32 +79,19 @@ public function __construct(
...
 			'admin' => $this->groupManager->isAdmin($this->userId),
-			'url' => $documentationUrl,
 			'urlUserDocs' => $urlUserDocs,
 			'urlAdminDocs' => $urlAdminDocs,
-			'mode' => $mode,
 			'pageTitle' => $pageTitle,
 		]);
-		$policy = new ContentSecurityPolicy();
-		$policy->addAllowedFrameDomain('\'self\'');
-		$response->setContentSecurityPolicy($policy);
```
Comment (szaimen) [inline apps/settings/lib/Controller/HelpController.php:107]: basically revert and generate urls based on config switch

File: apps/settings/templates/help.php [hunk 5]
```diff
@@ -1,48 +1,28 @@
 <?php
 \OC_Util::addStyle('settings', 'help');
 ?>
-<div id="app-navigation" role="navigation" tabindex="0">
```
Comment (szaimen) [inline apps/settings/templates/help.php:4]: modify template based on config switch

File: core/doc/admin/index.html
Comment (szaimen) [inline core/doc/admin/index.html:1]: re-add

File: apps/settings/templates/help.php [hunk 6]
```diff
@@ -1,48 +1,28 @@
...
-	<?php } ?>
-
-		<li>
-			<a href="https://docs.nextcloud.com" class="icon-category-office" target="_blank" rel="noreferrer noopener">
-				<span class="help-list__text">
-					<?php p($l->t('Documentation')); ?> ↗
-				</span>
+			<a class="button" target="_blank" rel="noreferrer noopener"
+				href="<?php print_unescaped($_['urlUserDocs']); ?>">
+				<?php p($l->t('Account documentation')); ?>  ↗
```
Comment (szaimen) [inline apps/settings/templates/help.php:18]: should be before admin docs

**Review by susnux — APPROVED** [review]

**Comment by susnux** [comment]
Tested and works, just the button style is a bit weird. It is not consistent with other buttons, is it?

**Comment by szaimen** [comment]
> Tested and works, just the button style is a bit weird. It is not consistent with other buttons, is it?

I am not sure. Is there a better css class that I could use or do I need to adjust the styling manually?

**Review by JuliaKirschenheuter — APPROVED** [review]

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**